go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -568,6 +570,46 @@ func (p DoltDatabaseProvider) DropDatabase(ctx *sql.Context, name string) error
 	return nil
 }
 
+// findAndDeleteNestedDatabaseDir tries to find matching database against nested database.
+// This function returns whether the database was found and deleted successfully.
+func (p DoltDatabaseProvider) findAndDeleteNestedDatabaseDir(dbName, dbKey string) (bool, error) {
```

Comment:
`DoltDatabaseProvider` has a `dbLocations` field with a map of database names to `FileSys`. It feels like this logic could be a bit simpler (and safer maybe?) if we used that as the authoritative source for database locations instead of redoing work to guess directories and checking to see if it's the right one.

Reply:
Ah I see, that definitely makes it lot easier and safer.